src/protocol/mod.rs

```diff
@@ -514,6 +540,12 @@ mod load {
         Ok(result.into_iter().next().unwrap())
     }
 
+    fn step_with_marker(command_matcher: CommandMatcher, marker: Marker) -> Step {
```

Comment:
i assume you had to duplicate the function above because here it is in a different sub `mod`? i ran into this same issue at one point and wasn't sure the best way to share the function between both. seems like you could either add a larger surrounding `mod` (and move the function up a level), or import the function from the other one like `use super::mod_name::function_name`. not sure what the best rusty way to do it is, curious what you and @soenkehahn think though.

Reply:
I did not think of importing from a sibling module. Maybe I could move `step_with_marker` into a test helper module. I try not to worry too much about duplicating code in tests. But there is no reason for duplication if it is easy to avoid.